examples/gno.land/p/demo/membstore/membstore.gno

```diff
@@ -0,0 +1,190 @@
+	}
+
+	// Apply the options
+	for _, opt := range opts {
+		opt(m)
+	}
+
+	return m
+}
+
+func (m *MembStore) AddMember(member Member) error {
+	if !m.isCallerGOVDAO() {
+		return ErrNotGovDAO
+	}
+
+	// Check if the member exists
+	if m.IsMember(member.Address) {
+		return ErrAlreadyMember
+	}
+
+	// Add the member
+	m.members.Set(member.Address.String(), member)
+
+	// Update the total voting power
+	m.totalVotingPower += member.VotingPower
+
+	return nil
+}
+
+func (m *MembStore) UpdateMember(address std.Address, member Member) error {
+	if !m.isCallerGOVDAO() {
+		return ErrNotGovDAO
+	}
+
+	// Get the member
+	oldMember, err := m.Member(address)
+	if err != nil {
+		return err
+	}
+
+	// Check if this is a removal request
+	if member.VotingPower == 0 {
+		m.members.Remove(address.String())
+
+		// Update the total voting power
+		m.totalVotingPower -= oldMember.VotingPower
+
+		return nil
+	}
+
+	// Check that the member wouldn't be
+	// overwriting an existing one
+	isAddressUpdate := address != member.Address
+	if isAddressUpdate && m.IsMember(member.Address) {
+		return ErrInvalidAddressUpdate
+	}
+
+	// Remove the old member info
+	// in case the address changed
+	if address != member.Address {
+		m.members.Remove(address.String())
+	}
+
+	// Save the new member info
+	m.members.Set(member.Address.String(), member)
+
+	// Update the total voting power
+	difference := member.VotingPower - oldMember.VotingPower
+	m.totalVotingPower += difference
+
+	return nil
+}
+
+func (m *MembStore) IsMember(address std.Address) bool {
+	_, exists := m.members.Get(address.String())
+
+	return exists
+}
+
+func (m *MembStore) Member(address std.Address) (Member, error) {
+	member, exists := m.members.Get(address.String())
+	if !exists {
+		return Member{}, ErrMissingMember
+	}
+
+	return member.(Member), nil
+}
+
+func (m *MembStore) Members(offset, count uint64) []Member {
+	// Calculate the left and right bounds
+	if count < 1 || offset >= uint64(m.members.Size()) {
+		return []Member{}
+	}
+
+	// Limit the maximum number of returned members
+	if count > maxRequestMembers {
+		count = maxRequestMembers
+	}
+
+	// Gather the members
+	members := make([]Member, 0)
+	m.members.IterateByOffset(
+		int(offset),
+		int(count),
+		func(_ string, val interface{}) bool {
+			member := val.(Member)
+
+			// Save the member
+			members = append(members, member)
+
+			return false
+		})
+
+	return members
+}
+
+func (m *MembStore) Size() int {
+	return m.members.Size()
+}
+
+// TotalPower returns the total voting power
+// of the member store
+func (m *MembStore) TotalPower() uint64 {
+	return m.totalVotingPower
+}
+
+// isCallerGOVDAO returns a flag indicating if the
+// current caller context is the active GOVDAO.
+// We need to include a govdao guard, even if the
+// executor guarantees it, because
+// the API of the member store is public and callable
+// by anyone who has a reference to the member store instance.
+func (m *MembStore) isCallerGOVDAO() bool {
```

Comment:
```suggestion
func (m *MembStore) isCallerDAORealm() bool {
```

Reply:
Please, remove the term “GovDAO” from p/ generic pure libraries.

Reply:
Renamed:

`[852291c](https://github.com/gnolang/gno/pull/2581/commits/852291cb342d7eacdce178ae2f24084767a77126)`